### PR TITLE
cleanup(flag): remove old `AROptionsNewSalePage` flag and code

### PR DESCRIPTION
**Pull request by pvinis**
Co-authored-by: Brian Beckerle <brian.beckerle@artsymail.com>

The type of this PR is: **cleanup**

### Description

This flag is pretty old and is actually broken. Turning off shows a react error for unregistered component
when navigating to an auction.

Paired with @brainbicycle to remove the old flag for auctions.

🧹 `old code` 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Remove old auction code and flag - pavlos

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ✅ **No changelog changes**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1bf0f4831e2d0195e4de165cb3811320a3031c97
</p>

**Review by brainbicycle — APPROVED** [review]
thanks !

**Review by MounirDhahri — COMMENTED** [review]
Thanks 🙌 cc @rajsam003 since I know you were working on this

File: src/app/AppRegistry.tsx [hunk 1]
```diff
@@ -327,7 +327,6 @@ export const modules = defineModules({
   ArtworkMedium: reactModule(ArtworkMediumQueryRenderer),
   ArtworkAttributionClassFAQ: reactModule(ArtworkAttributionClassFAQQueryRenderer),
   ArtworkSubmissionStatusFAQ: reactModule(ArtworkSubmissionStatusFAQ),
-  Auction: nativeModule(),
   Auction2: reactModule(SaleQueryRenderer, { fullBleed: true }, SaleScreenQuery),
```
[the remark this answers is not in the document]
Comment (rajsam003) [reply src/app/AppRegistry.tsx:330]: @MounirDhahri , There is already [PR](https://github.com/artsy/eigen/pull/6326) for this, which is already merged. 🚤

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/AppRegistry.tsx:330]: cool, thanks

**Pull request merged**